hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -60,7 +60,7 @@ public static String getRecordKeyFromGenericRecord(GenericRecord genericRecord,
    * @return the partition path for the passed in generic record.
    */
   public static String getPartitionPathFromGenericRecord(GenericRecord genericRecord, Option<BaseKeyGenerator> keyGeneratorOpt) {
-    return keyGeneratorOpt.isPresent() ? keyGeneratorOpt.get().getRecordKey(genericRecord) : genericRecord.get(HoodieRecord.PARTITION_PATH_METADATA_FIELD).toString();
+    return keyGeneratorOpt.isPresent() ? keyGeneratorOpt.get().getPartitionPath(genericRecord) : genericRecord.get(HoodieRecord.PARTITION_PATH_METADATA_FIELD).toString();
```

Comment:
this was never invoked for COW table and hence didn't hit this issue.

Reply:
@nsivabalan Got it. can you double check this again across the board. to see no typos.

Reply:
yes, I have tests succeeding at writeClient, spark datasource and deltastreamer as well for both COW and MOR.